### PR TITLE
Add Simple Moving Average (SMA) Calculation

### PR DESCRIPTION
**Pull request by kelork**
This commit adds a Python script for calculating the Simple Moving Average (SMA) of a time series data.

The script also includes a doctest that verifies the correctness of the SMA calculations for a sample dataset.

Usage:
- Run the script with your own time series data and specify the window size for SMA calculations.

### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Comment by kelork** [comment]
Hey can any of the admins review this, all tests have been passed.

File: financial/simple_moving_average.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    :param data: A list of numerical data points.
+    :param window_size: An integer representing the size of the SMA window.
+    :return: A list of SMA values with the same length as the input data.
+
+    The Simple Moving Average (SMA) is a statistical calculation used to
+    analyze data points by creating
+    a constantly updated average price over a specific time period.
+    In finance, SMA is often used in technical
+    analysis to smooth out price data and identify trends.
```
Comment (tianyizheng02) [inline financial/simple_moving_average.py:19]:
```suggestion
    Calculate the simple moving average (SMA) for some given time series data.

    :param data: A list of numerical data points.
    :param window_size: An integer representing the size of the SMA window.
    :return: A list of SMA values with the same length as the input data.
```

Let's keep the simple function explanation in the docstring and have the more in-depth explanation of the algorithm at the top of the file.

File: financial/simple_moving_average.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
+"""
+Calculate the Simple Moving Average (SMA) for a time series data.
+https://en.wikipedia.org/wiki/Moving_average
```
Comment (tianyizheng02) [inline financial/simple_moving_average.py:3]:
```suggestion
The Simple Moving Average (SMA) is a statistical calculation used to analyze data points
by creating a constantly updated average price over a specific time period. In finance, SMA is 
often used in time series analysis to smooth out price data and identify trends.

Reference: https://en.wikipedia.org/wiki/Moving_average
```

The more in-depth explanation can go here

File: financial/simple_moving_average.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+"""
+
+
+def simple_moving_average(data: list[int], window_size: int) -> list[float | None]:
```
Comment (tianyizheng02) [inline financial/simple_moving_average.py:7]:
```suggestion
def simple_moving_average(data: list[float], window_size: int) -> list[float | None]:
```
SMA also works with non-integer numerical data

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]

File: financial/simple_moving_average.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    :param window_size: An integer representing the size of the SMA window.
+    :return: A list of SMA values with the same length as the input data.
+
+    Example:
+    >>> sma = simple_moving_average([10, 12, 15, 13, 14, 16, 18, 17, 19, 21], 3)
+    >>> [round(value, 2) if value is not None else None for value in sma]
+    [None, None, 12.33, 13.33, 14.0, 14.33, 16.0, 17.0, 18.0, 19.0]
+    """
+
+    sma: list[float | None] = []
```
Comment (tianyizheng02) [inline financial/simple_moving_average.py:25]:
```suggestion
    Examples:
    >>> sma = simple_moving_average([10, 12, 15, 13, 14, 16, 18, 17, 19, 21], 3)
    >>> [round(value, 2) if value is not None else None for value in sma]
    [None, None, 12.33, 13.33, 14.0, 14.33, 16.0, 17.0, 18.0, 19.0]
    >>> simple_moving_average([10, 12, 15], 5)
    [None, None, None]
    >>> simple_moving_average([10, 12, 15, 13, 14, 16, 18, 17, 19, 21], 0)
    Traceback (most recent call last):
    ...
    ValueError: Window size must be a positive integer
    """
    if window_size < 1:
        raise ValueError("Window size must be a positive integer")

    sma: list[float | None] = []
```
Added test cases for when the window size is larger than the list and when the window size is invalid

**Review by tianyizheng02 — APPROVED** [review]

**Comment by tianyizheng02** [comment]
Thanks for your contribution!

**Comment by kelork** [comment]
You're welcome! I'm happy to be able to contribute.

**Pull request merged**